packages/autocomplete/src/containers/AutocompleteContainer.js

```diff
@@ -0,0 +1,540 @@
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.ENTER]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        if (this.focusSelectionModel.hasSelection()) {
+          this.selectItem(this.indexKeyMap[this.focusSelectionModel.selectedIndex]);
+        } else if (Object.keys(this.indexKeyMap).length > 0) {
+          this.selectItem(this.indexKeyMap[0]);
+        }
+
+        e.preventDefault();
+      } else if (!this.tagSelectionModel.hasSelection()) {
+        this.openDropdown();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.DOWN]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectNext();
+        e.preventDefault();
+      } else {
+        this.focusSelectionModel.selectFirst();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.UP]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectPrevious();
+        e.preventDefault();
+      } else {
+        this.focusSelectionModel.selectLast();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.SPACE]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (!isOpen) {
+        this.openDropdown();
+        e.preventDefault();
+      }
+    }
+  };
+
+  getItemId = key =>
+    typeof key === 'undefined' ? '' : `${this.getControlledState().id}--item-${key}`;
+
+  getTagId = key =>
+    typeof key === 'undefined' ? '' : `${this.getControlledState().id}--tag-${key}`;
+
+  getMenuId = () => `${this.getControlledState().id}--menu`;
+
+  getInputProps = ({ tabIndex = 0, onChange, onKeyDown, onBlur, ...other } = {}) => {
+    const { focusedKey, tagFocusedKey, isOpen } = this.getControlledState();
+
+    return {
+      tabIndex,
+      'aria-autocomplete': 'list',
+      'aria-activedescendant': isOpen ? this.getItemId(focusedKey) : this.getTagId(tagFocusedKey),
+      autoComplete: 'off',
+      onBlur: composeEventHandlers(onBlur, () => {
+        if (!this.menuMousedDown && !this.triggerMousedDown) {
+          this.closeDropdown();
+        }
+      }),
+      onChange: composeEventHandlers(onChange, () => {
+        this.focusSelectionModel.clearSelection();
+      }),
+      onKeyDown: composeEventHandlers(onKeyDown, e => {
+        const keydownHandler = this.INPUT_KEYDOWN_HANDLERS[e.keyCode];
+
+        keydownHandler && keydownHandler(e);
+      }),
+      ...other
+    };
+  };
+
+  getTriggerProps = ({ onMouseDown, onMouseUp, onClick, ...other }) => {
+    return {
+      onMouseDown: composeEventHandlers(onMouseDown, () => {
+        this.triggerMousedDown = true;
+      }),
+      onMouseUp: composeEventHandlers(onMouseUp, () => {
+        this.triggerMousedDown = false;
+      }),
+      onClick: composeEventHandlers(onClick, () => {
+        this.openDropdown();
+        this.inputRef.focus();
+      }),
+      ...other
```

Comment:
According the the authoring practices we need to add some more stuff here:

```
role: 'combobox',
'aria-expanded': isOpen, // may nee to pass strings as false will just remove this
'aria-owns': this.getMenuId(),
'aria-haspopup': 'listbox'
```

Reply:
Was looking at the wrong pattern for listbox when I implement this 😱 

I'm now fully matching the pattern at https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.0pattern/combobox-autocomplete-list.html